### PR TITLE
Properly migrate target branch change GitLab comment

### PR DESCRIPTION
**Pull request by invliD**
GitLab generates "system notes" whenever an event happens within the platform. Unlike Gitea, those events are stored and retrieved as text comments with no semantic details. The only way to tell whether a comment was generated in this manner is the `system` flag on the note type.

This PR adds detection for a new specific kind of event: Changing the target branch of a PR. When detected, it is downloaded using Gitea's type for this event, and eventually uploaded into Gitea in the expected format, i.e. with no text content in the comment.

This PR also updates the template used to render comments to add support for migrated comments of this type.

ref: https://gitlab.com/gitlab-org/gitlab/-/blob/11bd6dc826e0bea2832324a1d7356949a9398884/app/services/system_notes/merge_requests_service.rb#L102

### CODE REVIEW

File: templates/repo/issue/view_content/comments.tmpl [hunk 1]
```diff
@@ -495,9 +495,22 @@
 		{{else if eq .Type 25}}
 			<div class="timeline-item event">
 				<span class="badge">{{svg "octicon-git-branch"}}</span>
-				{{template "shared/user/avatarlink" dict "user" .Poster}}
+				{{if .OriginalAuthor}}
+				{{else}}
+					{{template "shared/user/avatarlink" dict "user" .Poster}}
+				{{end}}
 				<span class="text grey muted-links">
-					<a{{if gt .Poster.ID 0}} href="{{.Poster.HomeLink}}"{{end}}>{{.Poster.Name}}</a>
```
Comment (invliD) [inline templates/repo/issue/view_content/comments.tmpl:500]: I made a small drive-by change here and replaced this line with `{{template "shared/user/authorlink" .Poster}}` (line R512), as that's how authors are linked elsewhere in this file.

**Review by lunny — APPROVED** [review]

File: services/migrations/gitea_uploader.go [hunk 2]
```diff
@@ -487,6 +487,12 @@ func (g *GiteaLocalUploader) CreateComments(comments ...*base.Comment) error {
 			if comment.Meta["NewTitle"] != nil {
 				cm.NewTitle = fmt.Sprintf("%s", comment.Meta["NewTitle"])
 			}
+		case issues_model.CommentTypeChangeTargetBranch:
+			if comment.Meta["OldRef"] != nil && comment.Meta["NewRef"] != nil {
+				cm.OldRef = fmt.Sprintf("%s", comment.Meta["OldRef"])
+				cm.NewRef = fmt.Sprintf("%s", comment.Meta["NewRef"])
```
Comment (wxiaoguang) [inline services/migrations/gitea_uploader.go]:
```suggestion
				cm.OldRef = fmt.Sprint(comment.Meta["OldRef"])
				cm.NewRef = fmt.Sprint(comment.Meta["NewRef"])
```

File: templates/repo/issue/view_content/comments.tmpl [hunk 3]
```diff
@@ -495,9 +495,22 @@
 		{{else if eq .Type 25}}
 			<div class="timeline-item event">
 				<span class="badge">{{svg "octicon-git-branch"}}</span>
-				{{template "shared/user/avatarlink" dict "user" .Poster}}
+				{{if .OriginalAuthor}}
+				{{else}}
```
Comment (wxiaoguang) [inline templates/repo/issue/view_content/comments.tmpl]: Why an empty block?

* If it needs to show something ... then show something
* If it doesn't need to show anything, it could be `{{if not ...}}` (and maybe add some comments)

Comment (invliD) [reply templates/repo/issue/view_content/comments.tmpl]: I copied this from elsewhere in this file and didn't want to break convention. I'll update it here.

**Review by wxiaoguang — APPROVED** [review]
